src/Mapping/Driver/AttributeDriver.php

```diff
@@ -47,8 +51,19 @@ public function __construct(array $paths, bool $reportFieldsWhereDeclared = true
             ));
         }
 
+        if ($pathsAsSourceFilePathNames) {
+            if (! property_exists(static::class, 'sourceFilePathNames')) {
+                throw new LogicException('Source file path names support for AttributeDriver is available since doctrine/persistence 4.1.');
+            }
+
+            $this->sourceFilePathNames = $paths;
+        } elseif (is_array($paths)) {
+            $this->addPaths($paths);
+        } else {
+            throw new LogicException('The $paths argument must be an iterable of source file path names or an array of directories.');
```

Comment:
We're in the case where `$pathsAsSourceFilePathNames` is `false`


```suggestion
            throw new LogicException('The $paths argument must be an array of directories.');
```

Reply:
Yes, we are.
Though I think it would make sense to point out to the client that there are two ways of achieving the end goal, not just an array of directories.

Reply:
Maybe, but right now they might try it and not notice they also need to flip `$pathsAsSourceFilePathNames` and file a bug report.